src/Geometry.h

```diff
@@ -135,17 +136,16 @@ class Geometry {
   Point* getFSRPoint(int fsr_id);
   std::string getFSRKey(LocalCoords* coords);
 #ifndef CUDA
-  std::unordered_map<std::size_t, fsr_data> getFSRKeysMap();
+  ParallelHashMap<std::size_t, fsr_data*>* getFSRKeysMap();
 #endif
 
   /* Set parameters */
   void setFSRsToMaterialIDs(std::vector<int> FSRs_to_material_IDs);
   void setFSRsToKeys(std::vector<std::size_t> FSRs_to_keys);
-  void setNumFSRs(int num_fsrs);
   void setCmfd(Cmfd* cmfd);
 
 #ifndef CUDA
-  void setFSRKeysMap(std::unordered_map<std::size_t, fsr_data> FSR_keys_map);
+  void setFSRKeysMap(ParallelHashMap<std::size_t, fsr_data*>* FSR_keys_map);
```

Comment:
And if you see any other routines here that are public out of necessity but which should not be exposed to the user in Python, go ahead and ``%ignore`` them as well (including perhaps ``Geometry::getFSRKeysMap()``).